### ISSUE TITLE
[Issue/Bug]: Editor Presence causes Godot to force close on project startup

### ISSUE THREAD

**Issue by Faeest** [vaporvee/discord-rpc-godot#77]
### What happened?

When I first installed the Discord RPC plugin by VaporVee and enabled the editor presence feature, it worked fine. but when I tried to do it the next day Godot would crash and force close as soon as I opened the project. 

To resolve this issue, I investigated the plugin's code and found that the error seemed to be caused by the plugin trying to add a child node to the scene tree before the editor was fully ready. 

As a workaround, I edited the plugin's code to introduce a 30-second delay before adding the child node to the scene tree. This delay allowed the editor to fully initialize, and it resolved the crashing issue for me. 

However, I then noticed that the plugin's script was not correctly removing the "DiscordRPC/EditorPresence/enabled" setting when the project was closed or the scene tree was in the process of exiting. This meant that I had to manually toggle the setting off and on again to get the plugin working properly. 

I'm not sure why the setting is not being removed as expected, even though the plugin appears to have the necessary code to handle this. This is an additional issue I'm facing with the plugin that I hope can be resolved.

### Version

1.3.1

### Godot Version

4.2.2.stable

### Exact steps to reproduce this error

1. Create a new Godot project.
2. Install the Discord RPC plugin by VaporVee and enable it.
3. Go to the editor settings and enable the "DiscordRPC/EditorPresence/enabled" setting.
4. Save the project and restart Godot.
5. Godot will crash and force close as soon as the project loads.

### GDScript

```gdscript
if get_editor_interface().get_editor_settings().has_setting("DiscordRPC/EditorPresence/enabled") && get_editor_interface().get_editor_settings().get_setting("DiscordRPC/EditorPresence/enabled"):
		await get_tree().create_timer(30).timeout # <----------- my code
		add_child(editor_presence)
```

### Godot output

```shell
it works
```

### Checks

- [X] I tried reinstalling the addon or tried to fix it myself with other methods.
- [X] I tried restarting Discord and Godot completely.
- [X] I did read the documentation and its troubleshooting guide https://docs.vaporvee.com/discord-rpc-godot/

**Comment by vaporvee** [comment]
yeah thats the issue its not really possible to know when the editor is fully ready. do you think to add a timeout is a good idea in main? I don't know if people care for the delay because on some machines Godot loads fast enough and then the addon can enable that early without any problems. I need to investigate if there is a way to get a better signal for the editor beeing fully ready.

**Comment by vaporvee** [comment]
If there is no good signal for that i might need to do a Godot pr

**Comment by Faeest** [comment]
I think that'll do it

**Comment by Pab10Suarez** [comment]
i dont know if this helps, but i found the same bug and it only happens when you save the project and restart godot

**Comment by Pab10Suarez** [comment]
also i guess it depends on the project but i tried your work around @Faeest and put only a second and the plugin now works great so thanks!
